hashin.py

```diff
@@ -773,6 +773,23 @@ def main():
     parser = get_parser()
     args = parser.parse_args()
 
+    if (
+        args.update_all
+        and args.packages
+        and len(args.packages) == 1
+        and os.path.isfile(args.packages[0])
+        and args.packages[0].endswith(".txt")
```

Comment:
I think it would a small improvement to re-arrange these so that the file existence check happens last, since that is going to be the most expensive one.